aten/src/ATen/native/DilatedConvolution.cpp

```diff
@@ -0,0 +1,879 @@
+
+#include <ATen/div_rtn.h>
+#include <algorithm>
+#include <tuple>
+#include "ATen/ATen.h"
+#include "ATen/native/im2col.h"
+#include "ATen/native/vol2col.h"
+#include "TH/THBlasUtils.h"
+
+#define TORCH_CHECK_DIM_SIZE(T, DIM, DIM_SIZE, SIZE) \
+  TORCH_CHECK(                                       \
+      T.dim() == DIM && T.size(DIM_SIZE) == SIZE,    \
+      "Need " #T " of dimension ",                   \
+      DIM,                                           \
+      " and " #T ".size[",                           \
+      DIM_SIZE,                                      \
+      "] == ",                                       \
+      SIZE,                                          \
+      " but got input to be of shape ",              \
+      T.sizes())
+
+/*
+  Some convenience macros
+ */
+
+#define INSERT_BATCH_DIMENSION(A, SIZE)        \
+  if (A.defined()) {                           \
+    auto new_sizes = A.sizes().vec();          \
+    new_sizes.insert(new_sizes.begin(), SIZE); \
+    A.resize_(new_sizes);                      \
+  }
+
+#define DROP_BATCH_DIMENSION(A)    \
+  if (A.defined()) {               \
+    A.resize_(A.sizes().slice(1)); \
+  }
+
+namespace at {
+namespace native {
+
+namespace {
+
+inline bool all_positive(IntArrayRef& arr) {
+  return std::all_of(
+      arr.begin(), arr.end(), [](int64_t item) { return item > 0; });
+}
+
+#define OUTPUTSIZE(INDEX, DIM)                                       \
+  (input.size((INDEX) + input.dim() - (DIM)) + 2 * pad_size[INDEX] - \
+   (dilation_size[INDEX] * (kernel_size[INDEX] - 1) + 1)) /          \
+          stride_size[INDEX] +                                       \
+      1
+
+void conv_dilated_shape_check(
+    int64_t dim,
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& grad_output,
+    const Tensor& grad_input,
+    const Tensor& grad_weight,
+    const Tensor& grad_bias,
+    IntArrayRef kernel_size,
+    IntArrayRef stride_size,
+    IntArrayRef pad_size,
+    IntArrayRef dilation_size,
+    const Tensor& columns) {
```

Comment:
Columns? :)

Reply:
(You got rid of user-passed columns, right? So no need to check; if you made it, it's gunna be contiguous)

Reply:
columns are now removed from check functions.